### PR TITLE
fix(Tools/Mapextractor): Fix water height redundancy algorithm ignoring "no water"

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/pull/26324
## Tests Performed:

- 
- 

## How to Test the Changes:

1. Need Re-Extract map/mmaps
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by IntelligentQuantum** [comment]
@pklloveyou Hey, can you test it with RecastDemo please

**Comment by pklloveyou** [comment]
ok

**Comment by pklloveyou** [comment]
I tried many places that are easy to make mistakes, and there were no mistakes.
I tested it by the way#109478
The extracted mmaps file is less

**Comment by Nefertumm** [comment]
> I tested it by the way #10948 I tried many places that are easy to make mistakes, and there were no mistakes. Fewer mmaps files were extracted than before

@pklloveyou thank you for testing, but I didn't understand if it worked or not.

File: src/tools/map_extractor/System.cpp [hunk 1]
```diff
@@ -841,7 +841,10 @@ bool ConvertADT(std::string const& inputPath, std::string const& outputPath, int
                     if (minHeight > h) minHeight = h;
                 }
                 else
+                {
                     liquid_height[y][x] = CONF_use_minHeight;
+                    if (minHeight > CONF_use_minHeight) minHeight = CONF_use_minHeight;
```
Comment (Winfidonarleyan) [inline src/tools/map_extractor/System.cpp:846]: Codestyle

**Comment by pklloveyou** [comment]
We can merge:
https://github.com/azerothcore/azerothcore-wotlk/pull/10841
https://github.com/azerothcore/azerothcore-wotlk/pull/10921
https://github.com/azerothcore/azerothcore-wotlk/pull/10922
Working properly on my server.

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Lynethris** [comment]
> .go c id 10184

I have done a test and I see any issue with the map or the boss. Can you specify exactly what do you see ? Maybe a screenshot.

**Comment by Lynethris** [comment]
Tested done.

- [X] No error during the map extraction
- [X] No glitch find with water
- [X] No issue during my travel around my travel in AZ
- [X] No issue find with Thunder bluff water puddle
- [X] Fix this TC issue (present also on AC) https://github.com/TrinityCore/TrinityCore/issues/20221

**Comment by IntelligentQuantum** [comment]
> @IntelligentQuantum Test done.
> 
>     * [x]  No error during the map extraction
> 
>     * [x]  No glitch find with water
> 
>     * [x]  No special issue during my fast travel in Azeroth
> 
>     * [x]  No issue find with Thunder bluff water puddle
> 
>     * [x]  Fix this TC issue (present also on AC) [[3.3.5] Breathing bar on land TrinityCore/TrinityCore#20221](https://github.com/TrinityCore/TrinityCore/issues/20221)

Thank you

**Pull request merged**